cassandra/policies.py

```diff
@@ -376,7 +380,36 @@ def make_query_plan(self, working_keyspace=None, query=None):
                 for host in child.make_query_plan(keyspace, query):
                     yield host
             else:
-                replicas = self._cluster_metadata.get_replicas(keyspace, routing_key)
+                if EXPERIMENTAL_TABLETS and self._cluster_metadata.use_tablets():
+                    tablets = self._cluster_metadata.all_tablets()
+
+                    l = -1
+                    for i, tablet in enumerate(tablets): 
+                        if tablet.keyspace_name == keyspace and tablet.table_name == query.table:
+                            l = i
+                            break
```

Comment:
It would have to be dict indexed by `(keyspace, table_name, last_token)` I believe, do you think it would be better even with three element tuple as an index?

Reply:
Hmm.. Then maybe I don't understand this code after all. Here's my understanding of what's going on here, is it correct?

`tablets` is a list of `Tablet` object, sorted by `(Tablet.keyspace_name, Tablet.table_id, Tablet.last_token)`.
First loop basically finds sublist of `tablets` that describes given keyspace and table.
It does that by iterating trough this whole list, to find the first element with matching keyspace name and table name. This element is the left border of our sublist.
Field `tablet_count` of this element gives us right border of sublist.
Then we perform binsearch on resulting sublist to find the tablet responsible for given token.

What I proposed is is to change (pseudocode types) `List[Tablet]` into `Map[(keyspace, table_name), List[Tablet]]` - so we can avoid the first loop, instead getting a list for given `(keyspace, table)` directly. We'd still perform binsearch on resulting list.
We could have a map with a key you described, but it'd have to be a tree map, not a hash map, in order to efficiently perform queries like `find first element with key not less than` - I don't think there is any in Python stdlib, and I'd be reluctant to add dependency for this.

Reply:
I misunderstood you, I agree with your proposal

Reply:
Done